docs/blog/2021-12-01-csp.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Don't trust your Content Security Policy
+description: Why the CSP most sites have is useless.
+slug: csp
+authors:
+- name: Forrest Allison
+  title: Developer at LunaSec
+  url: https://github.com/factoidforrest
+  image_url: https://github.com/factoidforrest.png
+  tags: [ci, github-actions, docker.sock]
+---
+<!--
+  ~ Copyright by LunaSec (owned by Refinery Labs, Inc)
+  ~
+  ~ Licensed under the Creative Commons Attribution-ShareAlike 4.0 International
+  ~ (the "License"); you may not use this file except in compliance with the
+  ~ License. You may obtain a copy of the License at
+  ~
+  ~ https://creativecommons.org/licenses/by-sa/4.0/legalcode
+  ~
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~
+-->
+
+I'm a normal, non-security focused developer, but I work for a security company.  I wasn't surprised to learn that our 
+services use a good, well-thought-out CSP to protect our users. 
+
+What did come as a *big* surprise to me is that most companies, even huge, well thought of firms, have CSPs that don't
+protect from *anything*, or don't even use them at all. For the most part, it's not their fault and there's **not much they can do
+about it.**  More on that below.
+
+### How CSPs work
+
+The Content Security Policy (CSP) is a response header or `<meta>` tag set by the server when a webpage first loads.  It controls 
```

Comment:
You should link out to this page to define CSP.

https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP